packages/teams-js/src/internal/communication.ts

```diff
@@ -267,21 +267,28 @@ function sendMessageToParentHelper(actionName: string, args: any[]): MessageRequ
   return request;
 }
 
+const processMessageLogger = communicationLogger.extend('processMessage');
+
 /**
  * @internal
  * Limited to Microsoft-internal use
  */
 function processMessage(evt: DOMMessageEvent): void {
   // Process only if we received a valid message
   if (!evt || !evt.data || typeof evt.data !== 'object') {
+    processMessageLogger('Unrecognized message format received by app, message being ignored');
     return;
   }
 
   // Process only if the message is coming from a different window and a valid origin
-  // valid origins are either a pre-known
+  // valid origins are either a pre-known origin or one specified by the app developer
+  // in their call to app.initialize
   const messageSource = evt.source || (evt.originalEvent && evt.originalEvent.source);
   const messageOrigin = evt.origin || (evt.originalEvent && evt.originalEvent.origin);
   if (!shouldProcessMessage(messageSource, messageOrigin)) {
+    processMessageLogger(
+      'Message being ignored by app because it is not coming from a different window with a valid origin',
```

Comment:
>a valid origin' [](http://example.com/codeflow?start=90&length=15)

Could we log the origin it is coming from here to help aid in debugging?

Reply:
(reading further, maybe this will be logged by some of the other log statements?)

Reply:
Yes, should be captured elsewhere